From f388794fa1b2acdfb14530f29018dcbbc12521a3 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Tue, 19 Mar 2024 18:40:01 +0100
Subject: [PATCH] misc: stm32_omm: disable OMM clock at runtime

OMM clock should not be enabled at runtime. Remove the dependency
between OMM driver and its children.

Change-Id: I81c7f96386649ec4be9ba382945fa7dc4ae5cc2b
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/369133
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Domain-Review: Patrice CHOTARD <patrice.chotard@foss.st.com>
---
 drivers/misc/stm32_omm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/misc/stm32_omm.c
+++ b/drivers/misc/stm32_omm.c
@@ -384,7 +384,7 @@ static int stm32_omm_probe(struct platfo
 		if (!child_access[i] || !of_device_is_available(omm->child_node[i]))
 			continue;
 
-		vdev = of_platform_device_create(omm->child_node[i], NULL, dev);
+		vdev = of_platform_device_create(omm->child_node[i], NULL, NULL);
 		if (!vdev) {
 			dev_err(dev, "Failed to create Octo Memory Manager child\n");
 			for (j = i; j > 0; --j) {
